lib/matplotlib/collections.py

```diff
@@ -856,36 +851,82 @@ def get_linewidth(self):
     def get_linestyle(self):
         return self._linestyles
 
-    def update_scalarmappable(self):
-        """Update colors from the scalar mappable array, if it is not None."""
+    def _set_mappable_flags(self):
+        edge0 = self._edge_is_mapped
+        face0 = self._face_is_mapped
         if self._A is None:
+            self._edge_is_mapped = False
+            self._face_is_mapped = False
+            # return False  # Nothing to map
```

Comment:
Delete? Returning early would be nice, but AFAICS the code may return True if `changed`.

Reply:
Deleted.  Returning there would require duplicating some of the checking code.  The commented line was left over from earlier experimentation.